src/repositories/donationRepository.ts

```diff
@@ -74,16 +76,23 @@ export const findDonationById = async (
     .getOne();
 };
 
-export const findUsersWhoDonatedToProject = async (
+export const findUsersWhoDonatedToProjectExcludeWhoLiked = async (
   projectId: number,
 ): Promise<{ walletAddress: string; email?: string }[]> => {
   return Donation.createQueryBuilder('donation')
+    .leftJoinAndSelect('donation.project', 'project')
     .leftJoin('donation.user', 'user')
+    .leftJoin(
+      Reaction,
+      'reaction',
+      'reaction.projectId = project.id AND user.id = reaction.userId',
+    )
     .distinctOn(['user.walletAddress'])
     .select('LOWER(user.walletAddress) AS "walletAddress", user.email as email')
-    .where(`"projectId"=:projectId`, {
+    .where(`donation."projectId"=:projectId`, {
       projectId,
     })
+    .andWhere(`reaction.id IS NULL`)
```

Comment:
Hmm not sure about this condition or the test it self. Will work on double checking this.

Reply:
Should it be instead:
```
    .andWhere(`reaction.userId != donation.userId`) 
```

Reply:
Or do you mean users who did not like the project?